Include/Rover/Basis.hpp

```diff
@@ -131,36 +345,60 @@ namespace Details {
           for(auto l = std::size_t(1); l < order.size() && !to_break; ++l) {
             const auto& second_arguments = trial[order[l]].m_arguments;
             visit_arguments([&](const auto& rhs, auto k) {
-              if(j == k && Details::solve_basis(result, m_to_negate[i + 1],
+              auto val = lhs;
+              if(j == k && Details::solve_basis(val, m_to_negate[i + 1],
                   lhs, rhs)) {
+                m_arguments.set_value(i, std::move(val));
                 to_break = true;
               }
             }, second_arguments);
           }
         }
       }, first_arguments);
-    }, m_elements.m_arguments);
+    }
     std::shuffle(order.begin(), order.end(), std::move(generator));
     const auto& first_result = trial[order[0]].m_result;
     for(auto l = std::size_t(1); l < order.size(); ++l) {
       const auto& second_result = trial[order[l]].m_result;
       if(first_result != second_result) {
-        Details::solve_basis(m_elements.m_result, m_to_negate[0], first_result,
+        Details::solve_basis(m_result, m_to_negate[0], first_result,
           second_result);
       }
     }
+    if(!factor_indexes.empty()) {
+      compute_factors(trial, factor_indexes);
+    }
   }
   
   template<typename S, typename T>
   typename Basis<S, T>::ScalarArguments Basis<S, T>::apply(const Arguments&
       arguments) const {
-    auto result = ScalarArguments(arguments_size(arguments));
+    auto result = ScalarArguments();
     visit_arguments([&](const auto& arg, auto i) {
-      visit_arguments([&](const auto& identity, auto j) {
+      m_arguments.visit([&](const auto& identity, auto j) {
         if(i == j) {
-          Details::apply_basis(result[i], m_to_negate[i + 1], arg, identity);
+          std::visit([&](const auto& val) {
+            if constexpr(is_factor_v<std::decay_t<decltype(val)>>) {
+              auto dimension = Details::find_dimension(val, arg);
+              if(!dimension) {
+                result.insert(result.end(), val.size() - 1,
+                  static_cast<ScalarType>(1.) / val.size());
+              } else {
+                for(auto i = std::size_t(1); i < val.size(); ++i) {
+                  if(i == *dimension) {
+                    result.push_back(static_cast<ScalarType>(1.));
```

Comment:
Can as well use `emplace_back`, but it won't be conventional where `ScalarType` is already `double`.